src/realm/impl/sequential_getter.hpp

```diff
@@ -64,8 +64,13 @@ class SequentialGetter : public SequentialGetterBase {
 
     REALM_FORCEINLINE bool cache_next(size_t index)
     {
-        // Return whether or not leaf array has changed (could be useful to know for caller)
-        if (index >= m_leaf_end || index < m_leaf_start) {
+        // Set m_leaf_ptr to point at the leaf that contains the value at column row `index`. Return whether or not
+        // the leaf has changed (could be useful to know for caller).
+
+        // FIXME: Below line has been commented away because array leafs might relocate during the lifetime of the 
+        // object that owns this SequentialGetter. Enable again when we have proper support for that.
+//        if (index >= m_leaf_end || index < m_leaf_start) 
```

Comment:
No, because it turned out that all users of SeqentialGetter just used it to fetch its m_leaf_ptr member and iterate through the leaf manually or call various find-methods on the leaf. 

So cache_next() is just called once for each 1000 items.

Class should probably be redesigned or renamed or simplified...